metricbeat/module/mongodb/collstats/collstats.go

```diff
@@ -32,74 +35,84 @@ func init() {
 	)
 }
 
-// MetricSet type defines all fields of the MetricSet
+// Metricset type defines all fields of the Metricset
 // As a minimum it must inherit the mb.BaseMetricSet fields, but can be extended with
 // additional entries. These variables can be used to persist data or configuration between
 // multiple fetch calls.
-type MetricSet struct {
-	*mongodb.MetricSet
+type Metricset struct {
+	*mongodb.Metricset
 }
 
-// New creates a new instance of the MetricSet
+// New creates a new instance of the Metricset
 // Part of new is also setting up the configuration by processing additional
 // configuration entries if needed.
 func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
-	ms, err := mongodb.NewMetricSet(base)
+	ms, err := mongodb.NewMetricset(base)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("could not create mongodb metricset: %w", err)
 	}
-	return &MetricSet{ms}, nil
+
+	return &Metricset{ms}, nil
 }
 
 // Fetch methods implements the data gathering and data conversion to the right
 // format. It publishes the event which is then forwarded to the output. In case
 // of an error set the Error field of mb.Event or simply call report.Error().
-func (m *MetricSet) Fetch(reporter mb.ReporterV2) error {
-	// instantiate direct connections to each of the configured Mongo hosts
-	mongoSession, err := mongodb.NewDirectSession(m.DialInfo)
+func (m *Metricset) Fetch(reporter mb.ReporterV2) error {
+	client, err := mongodb.NewClient(m.Metricset.Config, m.Module().Config().Timeout, 0)
 	if err != nil {
-		return errors.Wrap(err, "error creating new Session")
+		return fmt.Errorf("could not create mongodb client: %w", err)
 	}
-	defer mongoSession.Close()
 
-	result := mapstr.M{}
+	defer func() {
+		_ = client.Disconnect(context.Background())
+	}()
```

Comment:
This may be an unpopular opinion, but I think assigning to the blank identifier to satisfy the linter is an antipattern; it does not handle the error but also evades the rationale behind enforcing `require-explanation: true` and makes it harder in the future to clean up linter supression because it is harder to find and reason about.

Reply:
+1, I'd rather see an explanation of why errors can be ignored where possible using the `//nolint:errcheck $reason` comment instead.

Reply:
On top of my head, most of `Close` and `Disconnect` operations across metricbeat modules ignores the error. As the person who has tried to maintain this module on the last years, I just wanted to do the dirty work of upgrading the library and leave improvements to the @elastic/obs-service-integrations team now

Keep in mind that it takes me more time to write this message on GH than to write a line that logs the error.